tests/auth.test.ts

```diff
@@ -97,6 +98,110 @@ describe('account registration', () => {
     verify(emailService.sendEmailVerification(user.email, user.firstName, anyString()))
       .never();
   });
+
+  test('User cannot register with a handle that is already in use', async () => {
+    const conn = await DatabaseConnection.get();
+    const existingMember = UserFactory.fake();
+
+    await new PortalState()
+      .createUsers(existingMember)
+      .write();
+
+    const user = {
+      email: 'acm@ucsd.edu',
+      firstName: 'ACM',
+      lastName: 'UCSD',
+      password: 'password',
+      major: UserFactory.major(),
+      graduationYear: UserFactory.graduationYear(),
+      handle: existingMember.handle,
+    };
```

Comment:
nit: I think we should implement a `UserRegistrationFactory` to generate these fields for us, so that the tests are shorter and make it more clear which fields we are intentionally setting vs. which ones are arbitrary. E.g., every field in this object doesn't matter except handle, so we should have something instead like:
```ts
const user = UserRegistrationFactory.fake({ handle: existingMember.handle });
```

Reply:
This will also be useful for future test cases since we haven't fully written out our user registration test suite yet, so those tests would use also use this factory.